modules/server_processes_manager/php/mriuploadserverprocess.class.inc

```diff
@@ -130,10 +137,14 @@ class MriUploadServerProcess extends AbstractServerProcess
      */
     public function getShellCommand()
     {
-        return "source {$this->_mriCodePath}/" . self::ENVIRONMENT_BASENAME . " ; "
-               . "{$this->_mriCodePath}/" . self::IMAGING_UPLOAD_FILE_PATH
-               . " -upload_id "    . escapeshellarg($this->_mriUploadId)
-               . " -profile prod " . escapeshellarg($this->_sourceLocation);
+        // Sanitize variables before using in exec statement.
+        $codePath       = escapeshellarg($this->_mriCodePath);
```

Comment:
I think this is wrong. This variable seems to be used to construct another path (or two), and the whole argument(s) needs to be escaped after concatenation of the relevant parts, just just a subset of it.